rskj-core/src/main/java/co/rsk/core/bc/BlockExecutorFactory.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.core.bc;
+
+import co.rsk.config.RskSystemProperties;
+import co.rsk.db.StateRootHandler;
+import org.ethereum.core.BlockFactory;
+import org.ethereum.core.Repository;
+import org.ethereum.core.TransactionExecutor;
+import org.ethereum.db.BlockStore;
+import org.ethereum.db.ReceiptStore;
+import org.ethereum.listener.EthereumListener;
+import org.ethereum.vm.PrecompiledContracts;
+import org.ethereum.vm.program.invoke.ProgramInvokeFactory;
+import org.ethereum.vm.trace.ProgramTraceProcessor;
+
+/**
+ * Created by ajlopez on 23/04/2019.
+ */
+public class BlockExecutorFactory {
+    private final RskSystemProperties config;
+    private final Repository repository;
+    private final StateRootHandler stateRootHandler;
+    private final BlockStore blockStore;
+    private final ReceiptStore receiptStore;
+    private final ProgramInvokeFactory programInvokeFactory;
+    private final EthereumListener ethereumListener;
+    private final BlockFactory blockFactory;
+
+    public BlockExecutorFactory(RskSystemProperties config,
```

Comment:
Could we implement the factory pattern with a Lambda and an interface as it's done for other classes such as `RskWireProtocol.Factory`? I think it greatly reduces boilerplate code like the one introduced by this class.

Reply:
I think the current code is clear and explicit